bazel/repositories_extra.bzl

```diff
@@ -15,13 +15,17 @@ def _python_deps():
         requirements = "@envoy//configs:requirements.txt",
     )
     pip3_import(
-        name = "protodoc_pip3",
-        requirements = "@envoy//tools/protodoc:requirements.txt",
+        name = "kafka_pip3",
+        requirements = "@envoy//source/extensions/filters/network/kafka:requirements.txt",
```

Comment:
It seems like we added a hard dependency on the kafka extension.

Disabling this extension allowed you to continue to build even after deleting the contents of the source/extensions/filters/network/kafka directory.  That's no longer the case after this change.

Reply:
I don't know what the right solution is please work with @htuch 

If we had some way to query for enabled extensions in this file, we could wrap these import calls in an if so they only execute when the relevant extensions are enabled.

Reply:
I don't think the issue is whether the extension is enabled or not, in our case it is because we delete the extension tree during import. @antoniovicente do you think we could have Copybara preserve requirements.txt, even for extensions we don't care about?

If we want to condition on extension enablement, what I suggest is:
1. Add a per-extension `.bzl` for external dependencies that covers the pip3 business.
2. Add some generic conditional logic that skips loading it. But, I think it's hard to make loads condition in Bazel outside of WORKSPACE.

Reply:
The mechanism used to specify which extensions are enabled is part of what allows us to delete the directories associated with extensions since that mechanism controls which of the extension libraries are linked into the envoy binary.

I think we have an internal work-around for now so we can live with the current state for a while, but it would be good to think about a cleaner longer term direction.  Preserving requirements.txt for now seems like a possible option.